theano/sandbox/cuda/cusolver.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import absolute_import, print_function, division
+import pkg_resources
+
+import theano
+from theano.sandbox.cuda.type import CudaNdarrayType
+from theano.sandbox.cuda import CudaNdarray, GpuOp
+from theano.sandbox.cuda.basic_ops import as_cuda_ndarray_variable
+
+try:
+    from theano.sandbox.cuda import cuda_ndarray
+    dimshuffle = cuda_ndarray.cuda_ndarray.dimshuffle
+except ImportError:
+    pass
+
+cusolver_available = False
+
+try:
+    from scikits.cuda import cusolver
+    cusolver_available = True
+except (ImportError, OSError, RuntimeError, pkg_resources.DistributionNotFound):
+    pass
+
+cusolver_handle = [None]
+
+class GpuSolve(GpuOp):
```

Comment:
As we see there could be many implementation of that, I would prefer a more precise name. GpucuSolverXXX.